### PR TITLE
Add an interaction group to specify fluid/boundary interactions

### LINKED ISSUE #57: Filtering liquids with rapier rigidbodies based on masks/layers
Right now rapier has a lot of filtering going on. Salva liquids interact with any/all rigidbodies or static bodies from rapier.

Would be great if there was exposed a property in salva such that it only interacts with some rapier objects (eg masks and layers or something similar to what rapier has).

### PR DESCRIPTION
**Pull request by ThierryBerger**
- Fix https://github.com/dimforge/salva/issues/57

TODO:
- [ ] This PR doesn´t add the capacity to add interaction group between fluids (yet? It seems to be possible but didn´t look much into it yet.)
- [ ] check benchmarks

### CODE REVIEW

File: examples2d/layers2.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    let fluid_handle = fluids_pipeline.liquid_world.add_fluid(fluid);
+    plugin.set_fluid_color(fluid_handle, Point3::new(1.0, 0.4, 0.6));
+
+    let viscosity = ArtificialViscosity::new(0.5, 0.0);
+    let mut fluid = Fluid::new(
+        points3,
+        PARTICLE_RADIUS,
+        1.0,
+        InteractionGroups::none(),
+        //        InteractionGroups::new(Group::GROUP_3, Group::GROUP_3),
```
Comment (ThierryBerger) [inline examples2d/layers2.rs]: to remove

File: src/geometry/contacts.rs [hunk 2]
```diff
@@ -331,6 +341,13 @@ fn compute_contacts_for_pair_of_cells(
...
+                        let bj = &boundaries[fluid_j];
+                        if !fluids[*fluid_i]
+                            .boundary_interaction
+                            .test(bj.fluid_interaction)
+                        {
+                            continue;
+                        }
                         boundaries[fluid_j].positions[particle_j]
                     } else {
                         fluids[fluid_j].positions[particle_j]
```
Comment (sebcrozet) [inline src/geometry/contacts.rs:353]: There is a good chance that adding support for groups between two fluids is only a matter of adding a check in this branch too.

**Review by sebcrozet — APPROVED** [review]
Looks good, thank you!

**Comment by Ughuuu** [comment]
So this adds just between fluid/boundary? Still, thanks for this PR.

**Comment by ThierryBerger** [comment]
> So this adds just between fluid/boundary? Still, thanks for this PR.

Yes ; fluid / fluid layers will be studied in a follow up PR 🤞

**Pull request merged**